### ISSUE TITLE
Eventlistener unload and beforeunload perturbation with angular 2

### ISSUE THREAD

**Issue by natcohen** [Aymkdn/ExcelPlus#14]
My application is using the `onbeforeunload` listener to call whatever function before the webapp closes. Surprisingly, ExcelPlus perturbates this event and the `unload` event (maybe even more events - just not tested). Whenever I add ExcelPlus.js, this event is not fired anymore:

      window.onbeforeunload = (event)=> {
          console.log('before unload');
      }

It took me a lot of time to find out that this library is the origin of this bug...

**Comment by Aymkdn** [comment]
I don't see any reason for that... You use v2.4.1 ?
Did you try to remove files one by one, like js-xlsx too? 

There is nothing related to this event or to `window`

**Comment by natcohen** [comment]
Yes, 2.4.1...and yes, I've tried to remove one by one all my libraries and removing the minified version of ExcelPlus is what solved this issue.

Did you try?

**Comment by natcohen** [comment]
Oh wait... How can I remove xls-js? I only import sheetjs and ExcelPlus!

**Comment by Aymkdn** [comment]
I've just tried and it works as expected: https://jessy.kodono.info/test/ExcelPlus-gh-pages/demo/example3.html

If you look at the console you'll see the "before unload" (on refresh for example) and I use exactly your `window.onbeforeunload` code.

**Comment by natcohen** [comment]
I've succeeded to reproduce the problem: https://plnkr.co/edit/AFcVLVon59ABT0gVQyL6?p=preview (you need to refresh the page with and without excel-plus.js)

The problem seems to be the interaction between Angular 2 and ExcelPlus

**Comment by Aymkdn** [comment]
OK so the problem is due to:
```javascript
function addEventListener(el, eventName, handler) {
  if (el.addEventListener) {
    el.addEventListener(eventName, handler);
  } else if (el.attachEvent) {
    el.attachEvent('on' + eventName, function(){
      handler.call(el);
    });
  }
}
```

It's a polyfill that I used at the beginning of the file. You can rename this function to `excelPlus_addEventListener`, and it's used only once into my code so you can rename [line #176](https://github.com/Aymkdn/ExcelPlus/blob/master/2.4.1/excelplus-2.4.1.js#L176) with `excelPlus_addEventListener` too.

I'll correct that into the next release.

Thanks for bringing this issue to me!

**Comment by natcohen** [comment]
👍

**Comment by Aymkdn** [comment]
Fixed into the new release (v2.5)
